contracts/nois-gateway/src/msg.rs

```diff
@@ -1,10 +1,15 @@
 use cosmwasm_schema::{cw_serde, QueryResponses};
-use cosmwasm_std::HexBinary;
+use cosmwasm_std::{Coin, HexBinary};
 
 use crate::state::Config;
 
 #[cw_serde]
-pub struct InstantiateMsg {}
+pub struct InstantiateMsg {
+    /// The prices of a randomness. List is to be interpreted as oneof,
+    /// i.e. payment must be paid in one of those denominations.
+    pub prices: Vec<Coin>,
```

Comment:
I think here we don't need multiple options. In the proxy it makes sense as the proxy can convert itself. I'd just use this:

```suggestion
    pub price: Coin,
```

Reply:
We can keep it a list, no problem. Should not be much overhead